### ISSUE TITLE
[Regression] Visual Studio becomes unresponsive if you attempt open a bunch of C# files

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#37264]
**Version Used**: 
Version 16.3.0 Preview 2.0 [29111.228.master]

**Steps to Reproduce**:

-- Clone Repo --
1. git clone http://github.com/dotnet/project-system
2. `cd project-system`
3. `project-system> build`
4. `project-system> ProjectSystem.sln`

-- Open files --

4. `File` -> `Open`, navigate to the project-system folder above
5.. In Search, enter `ext:"cs"`
6. `CTRL+A` to select all, and choose `Open`

**Expected Behavior**:
No UI delays

**Actual Behavior**:
Visual Studio becomes unresponsible and never recovers

Trace is [internalshare]\davkean\Roslyn\36899.

**Comment by davkean** [comment]
All time is spent in here:

![image](https://user-images.githubusercontent.com/1103906/61274086-5a446e80-a7ee-11e9-9933-b24c77b1a29c.png)

**Comment by sharwell** [comment]
@davkean This is a JTF deadlock scenario. Do you have the stack trace for the other thread which is blocked waiting on the UI thread to proceed?

**Comment by davkean** [comment]
Is the main thread retrying?

**Comment by sharwell** [comment]
@davkean No, the main thread is blocked by effectively calling `Task<TResult>.Result`

**Comment by sharwell** [comment]
It's also possible the main thread is not deadlocked, and the process instead simply ran out of memory

**Comment by davkean** [comment]
The main thread is still pumping and chewing CPU.

**Comment by davkean** [comment]
Uploading a dump to the same share, but every time I break its on a different stack and not blocked. This is a 100% repro for me with above steps.
